integration-testing/src/functionalTest/kotlin/kotlinx.atomicfu.gradle.plugin.test/framework/runner/BuildRunner.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2016-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.atomicfu.gradle.plugin.test.framework.runner
+
+import java.io.File
+import java.nio.file.Files
+
+internal interface GradleBuild {
+    val projectName: String
+
+    val targetDir: File
+
+    var enableJvmIrTransformation: Boolean
+
+    var enableJsIrTransformation: Boolean
+
+    var enableNativeIrTransformation: Boolean
+
+    fun runGradle(commands: List<String>): BuildResult
+
+    fun clear()
+}
+
+private class GradleBuildImpl(
+    override val projectName: String,
+    override val targetDir: File
+) : GradleBuild {
+
+    override var enableJvmIrTransformation = false
+    override var enableJsIrTransformation = false
+    override var enableNativeIrTransformation = false
+
+    private val properties
+        get() = buildList {
+            add("-P$ENABLE_JVM_IR_TRANSFORMATION=$enableJvmIrTransformation")
```

Comment:
Unless there is an intention to have particular `kotlin_version` and `atomicfu_version` values in examples' properties, it's better to set them here as well. Otherwise, we're doomed to update it with every atomicfu release or Kotlin version update.

Reply:
Right, yeah, we can set these versions manually here, or just set them in `integration-testing/gradle.properties` file to the same values as in the root gradle.properties